File: unity_app/Assets/Scripts/Core/Enums/AchievementEnums.cs
namespace EventPlannerSim.Core
{
    /// <summary>
    /// Represents the type of achievement that can be earned.
    /// </summary>
    public enum AchievementType
    {
        // Progression
        FirstSteps,
        RisingStar,
        GoingPro,
        IndustryLeader,
        EventPlanningMogul,
        
        // Mastery
        PerfectPlanner,
        ConsistencyIsKey,
        ExcellenceStreak,
        BudgetMaster,
        VendorWhisperer,
        WeatherWatcher,
        
        // Challenge
        PerfectionistsPerfectionist,
        JugglingAct,
        CrisisManager,
        SelfMade,
        CorporateClimber,
        CelebrityHandler,
        
        // Secret
        AboveAndBeyond,
        FamilyFirst,
        ComebackKid
    }

    /// <summary>
    /// Represents the category of an achievement.
    /// </summary>
    public enum AchievementCategory
    {
        Progression,
        Mastery,
        Challenge,
        Secret
    }
}

File: EventApp/EventsApp/Models/Enums/AchievementEnums.swift
import Foundation

enum AchievementType: String, CaseIterable, Codable, Hashable {
    case firstSteps
    case risingStar
    case goingPro
    case industryLeader
    case eventPlanningMogul
    case perfectPlanner
    case consistencyIsKey
    case excellenceStreak
    case budgetMaster
    case vendorWhisperer
    case weatherWatcher
    case perfectionistsPerfectionist
    case jugglingAct
    case crisisManager
    case selfMade
    case corporateClimber
    case celebrityHandler
    case aboveAndBeyond
    case familyFirst
    case comebackKid
}

enum AchievementCategory: String, CaseIterable, Codable, Hashable {
    case progression
    case mastery
    case challenge
    case secret
}
